### PR TITLE
util: add check to verify that the job key sizes are at least the same

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
The PR contains updates for self-implemented function in verbs and sockets providers.
fi_tostr function is updated to print out the value of `auth_key_size` field of EP attributes.

### CODE REVIEW

**Comment by gladkovdmitry17** [comment]
@shefty @a-ilango Could you review, pls? Thanks in advance!

File: prov/verbs/src/verbs_info.c [hunk 1]
```diff
@@ -223,6 +223,14 @@ int fi_ibv_check_ep_attr(const struct fi_ep_attr *attr,
 		return -FI_ENODATA;
 	}
 
+	if (attr->auth_key_size &&
+	    (attr->auth_key_size != info->ep_attr->auth_key_size)) {
+		VERBS_INFO(FI_LOG_CORE, "Unsupported authentification size. "
+			   "Expected:%d, Requested%d\n",
+			   info->ep_attr->auth_key_size, attr->auth_key_size);
```
Comment (shefty) [inline prov/verbs/src/verbs_info.c]: Can the log messages above use FI_INFO_CHECK() from fi_util.h?  @a-ilango created that macro to provide consistency in the output.

Comment (a-ilango) [inline prov/verbs/src/verbs_info.c]: I believe that might not work for printing an ep_attr. It uses fi_tostr function which would print all attributes in ep_attr.

**Comment by shefty** [comment]
Got it - thanks.  Is there enough of a benefit for adding a macro for printing ints?  Though we can take that up at another time.

**Comment by a-ilango** [comment]
Yes, I think that would be useful. There are several places in check info function where we print ints.

**Comment by gladkovdmitry17** [comment]
@a-ilango @shefty I updated PR with proposed changes.

File: src/fi_tostr.c [hunk 2]
```diff
@@ -619,20 +621,27 @@ static void fi_tostr_cq_event_flags(char *buf, uint64_t flags)
...
+}
+
 __attribute__((visibility ("default")))
 char *DEFAULT_SYMVER_PRE(fi_tostr)(const void *data, enum fi_type datatype)
 {
 	static char *buf = NULL;
 	uint64_t val64;
 	uint32_t val32;
 	int enumval;
+	size_t valSizeT;
```
Comment (shefty) [inline src/fi_tostr.c]: You've been spending too much time in windows land.  :)  CamelCase isn't used elsewhere in the main portions of the project.

File: include/rdma/fabric.h [hunk 3]
```diff
@@ -546,6 +546,7 @@ enum fi_type {
 	FI_TYPE_CQ_EVENT_FLAGS,
 	FI_TYPE_MR_MODE,
 	FI_TYPE_OP_TYPE,
+	FI_TYPE_SIZE_T,
```
Comment (shefty) [inline include/rdma/fabric.h]: We don't want to expose basic data type through the API.

File: prov/verbs/src/verbs_info.c [hunk 4]
```diff
@@ -327,18 +340,21 @@ int fi_ibv_check_tx_attr(const struct fi_tx_attr *attr,
 	if (attr->size > info->tx_attr->size) {
 		VERBS_INFO(FI_LOG_CORE,
 			   "Given tx_attr->size is greater than supported\n");
+		FI_INFO_CHECK(&fi_ibv_prov, (info->tx_attr), attr, size, FI_TYPE_SIZE_T);
```
Comment (a-ilango) [inline prov/verbs/src/verbs_info.c]: Do we need the parenthesis around info->tx_attr? If it's needed it should be moved into the macro.

File: src/fi_tostr.c (hunk 2 above)
Comment (gladkovdmitry17) [inline src/fi_tostr.c]: yes :) will rename it

File: include/rdma/fabric.h (hunk 3 above)
Comment (gladkovdmitry17) [inline include/rdma/fabric.h]: maybe FI_TYPE_NUMERIC_FIELD will be okay? I just want to define common `fi_type` that can be used for numeric value like `max_msg_size` and etc (all of them has `size_t` datatype).

File: prov/verbs/src/verbs_info.c (hunk 4 above)
Comment (gladkovdmitry17) [inline prov/verbs/src/verbs_info.c]: I think, it will be more reasonable to moved into the macro.

**Comment by shefty** [comment]
These types are used to convert OFI specific structures and enums into a printable string.  We don't need this functionality for printing integers.  I would suggest having a new macro, like FI_INFO_CHECK_VAL, rather than modifying the external API so we can re-use an internal logging macro.

**Comment by gladkovdmitry17** [comment]
@shefty Thanks. It's pretty nice point. I re-implemented solution as you proposed adding new macro `FI_INFO_CHECK_VAL`.

**Comment by shefty** [comment]
thanks for sticking through with this

**Pull request merged**